content/docs/traffic-management/transformations/templating-language.md

```diff
@@ -0,0 +1,214 @@
+| `header(header_name)` | Returns the value of the header with the given name. For an example, see [Change response status](../change-response-status/). |
+| `raw_string(string)` | Returns the input string with escaped characters intact. This function is useful for constructing JSON request or response bodies. For an example, see [Inject response headers](../inject-response-headers). |
+| `replace_with_random(string, pattern)` | Finds the pattern in the input string and replaces this pattern with a random string. For an example, see [Inject response headers](../inject-response-headers/). |
+| `request_header(header_name)` | Returns the value of the request header with the given name. This function is useful to add the request header values in response transformations. For an example for how to extract a request header value and inject it into a response header, see [Inject response headers](../inject-response-headers/).|
+| `substring(string, start_pos, substring_len)` | Returns a substring of the input string, starting at `start_pos` and extending for `substring_len` characters. If no `substring_len` is provided or `substring_len is <= 0`, the substring extends to the end of the input string. For an example, see [Decode base64 headers](../decode-base64-headers/). |
+
+**Other common functions**
+
+You might use default Inja functions, such as `if else` or `if exists`. For an example, see [Change response status](../change-response-status/). 
+
+## Template attributes
+
+Learn more about the template attributes that you can use to transform headers and bodies of requests and responses.
+
+### `add`, `set`, `remove`
+
+Apply transformation templates to add, set, or remode request or response headers. 
+
+#### Static value
+
+You can add request and response headers with static values. The following example adds the `my-header: static` request header and sets the `foo: bar` response header. 
+
+```yaml
+transformation: 
+  request:
+    add: 
+    - name: my-header
+      value: static
+  response: 
+    set: 
+    - name: foo
+      value: bar
+```
+
+#### Inja functions
+
+You can use Inja functions in combination with Inja templates to modify request and response headers. 
+
+In the following example, you set response headers in the following ways: 
+* `x-gateway-response`: Use the value from the `x-gateway-request` request header and populate the value of that header into an `x-gateway-response` response header.
+* `x-podname`: Retrieve the value of the `POD_NAME` environment variable and add the value to the x-podname response header. Because the transformation is processed in the gateway proxy, these environment variables refer to the variables that are set on the proxy. You can view supported environment variables when you run `kubectl get deployment http -n kgateway-system -o yaml` and look at the `spec.containers.env` section.
```

Comment:
```suggestion
* `x-podname`: Retrieve the value of the `POD_NAME` environment variable and add the value to the x-podname response header. Because the transformation is processed in the gateway proxy, these environment variables refer to the variables that are set on the proxy. You can view supported environment variables when you run `kubectl get deployment http -n {{< reuse "/docs/snippets/namespace.md" >}} -o yaml` and look at the `spec.containers.env` section.
```